tests/rptest/tests/throughput_limits_snc_test.py

```diff
@@ -0,0 +1,180 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import random, time, math
+from enum import Enum
+from typing import Tuple
+from rptest.services.cluster import cluster
+from rptest.services.redpanda import MetricsEndpoint
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.clients.rpk import RpkTool
+from ducktape.tests.test import TestContext
+
+# This file is about throughput limiting that works at shard/node/cluster (SNC)
+# levels, like cluster-wide and node-wide throughput limits
+
+
+class ThroughputLimitsSnc(RedpandaTest):
+    """
+    Throughput limiting that works at shard/node/cluster (SNC)
+    levels, like cluster-wide and node-wide throughput limits
+    """
+
+    # see later if need to split into more classes
+
+    def __init__(self, test_ctx: TestContext, *args, **kwargs):
+        super(ThroughputLimitsSnc, self).__init__(test_ctx,
+                                                  num_brokers=3,
+                                                  *args,
+                                                  **kwargs)
+        rnd_seed = time.time()
+        rnd_seed_override = test_ctx.globals.get("random_seed", True)
+        if not rnd_seed_override is None:
```

Comment:
afaict rnd_seed_override will never be None.

Reply:
I stuggle to understand what type the `test_ctx.globals` is to verify your statement, however what I see in practice, if I don't supply the `random_seed` global to ducktape, I get `self.rnd_seed` initialized exactly in the way it's done in this `None` branch.

Reply:
it's a dictionary

Reply:
```
+        print(type(test_context.globals))
+        print(test_context.globals.get("lkjsdlfkjsdf", 12345))
+        print(test_context.globals.get("lkjsdlasdffffff"))

produces

<class 'ducktape.utils.persistence.PDict'>
12345
None
```

so i think you can drop the `, True` parameter. that is what was confusing.